engine/src/main/java/org/pentaho/di/trans/DefaultTransManager.java

```diff
@@ -0,0 +1,43 @@
+package org.pentaho.di.trans;
+
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.logging.LogChannelInterface;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class DefaultTransManager implements TransMananger {
+
+  private static DefaultTransManager instance;
+
+  private Map<String, TransFactory> mapConfigurationTypeTransFactory;
+
+
+  private DefaultTransManager(){
+    mapConfigurationTypeTransFactory = new HashMap<>();
+  }
+
+  public static synchronized DefaultTransManager getInstance() {
+    if ( null == instance ) {
+      instance = new DefaultTransManager();
+    }
+    return instance;
+  }
+
+  @Override
+  public Trans createTrans( TransMeta transMeta, LogChannelInterface log, String runConfiguration,
+    Supplier<Trans> fallbackSupplier ) throws KettleException {
+    TransFactory transFactory = getTransFactory( runConfiguration );
+    return transFactory.create( transMeta, log, fallbackSupplier );
```

Comment:
We are having am offline discussion on use use of `fallbackSupplier.` IMHO i think the use of `fallbacksupplier` is unnecessary design approach, back when the previous team did have ael as true plugin and ael was packaged along with the monolith pdi application.

We should assume that all the engine plugins have been registered and return null or throw an exception.

If we want to have some type of "default", then it should be hard coded in the  DefaultTransManager construction Or (brainstorming) engine plugins can signify that they will be the default, then you have to have sort of "tiebreaker" or fail it two plugins make themselves the "default"

Reply:
addressed